src/RenderWebGL.js

```diff
@@ -1002,10 +1002,18 @@ class RenderWebGL extends EventEmitter {
      * RenderConstants.ID_NONE if there is no Drawable at that location.
      */
     pick (centerX, centerY, touchWidth, touchHeight, candidateIDs) {
+        const bounds = this.clientSpaceToScratchBounds(centerX, centerY, touchWidth, touchHeight);
+        if (bounds.left === -Infinity || bounds.bottom === -Infinity) {
+            return false;
+        }
+
         candidateIDs = (candidateIDs || this._drawList).filter(id => {
             const drawable = this._allDrawables[id];
+            const drawableBounds = drawable.getFastBounds();
+            const inRange = bounds.intersects(drawableBounds);
             // default pick list ignores visible and ghosted sprites.
             if (drawable.getVisible() && drawable.getUniforms().u_ghost !== 0) {
```

Comment:
Move `drawable.getFastBounds` and `bounds.intersects(drawableBounds)` in here to avoid doing unnecessary work:

```js
if (drawable.getVisible() && drawable.getUniforms().u_ghost !== 0) {
    const drawableBounds = drawable.getFastBounds();
    const inRange = bounds.intersects(drawableBounds);
    if (!inRange) return false;
```

Reply:
done.